book.rb

```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```

Comment:
- You did a great effort by adding this method to add rental however I think this method cannot work because to add rental you need to pass into the params person and date. and also `the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.`  please follow this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) to learn more about the association :+1: 

```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

Reply:
ok changes made accordingly